src/ComponentRenderer.es6

```diff
@@ -15,18 +15,44 @@ class ComponentRenderer {
 
   attachToElement(element, state) {
     this.containerNode = element;
-    this.rerender(state);
+    this.state = state;
+    this.rerender();
   }
 
-  rerender(state) {
-    ReactDOM.render(this._componentFor(state), this.containerNode);
+  rerender() {
+    ReactDOM.render(this._componentFor(this.state), this.containerNode);
+  }
+
+  updateStateAndRerender(newState) {
+    if (newState.react.partial) {
+      delete newState.react.partial;
+      this.state = this._mergeRecursive(this.state, newState);
+    } else {
+      this.state = newState;
+    }
+    this.rerender();
   }
 
   _componentFor(state) {
-    const Component = this.components[state.component.name];
+    const Component = this.components[state.react.component];
     const factory = React.createFactory(Component);
     return factory(state);
   }
+
+  _mergeRecursive(oldObj, newObj) {
+    for (const k in newObj) {
+      try {
+        if (typeof newObj[k] == 'object' ) {
```

Comment:
To be perfectly honest this was a [copy/paste job](http://stackoverflow.com/a/383245/201130), but `===` seems like overkill as they're both strings; the exception catching is for the case where `newObj.key1` exists and is an object, but `oldObj` is not an object.